pkg/internal/nodeinfo/cpu_models.go

```diff
@@ -0,0 +1,333 @@
+func parseCpuString(cpuStr string) float64 {
+	if cpuStr == "" || cpuStr == "0" {
+		return 0.0
+	}
+
+	quantity, err := resource.ParseQuantity(cpuStr)
+	if err != nil {
+		return 0.0
+	}
+
+	return quantity.AsApproximateFloat64()
+}
+
+// parseMemoryString converts Kubernetes memory resource string to float64 (in GB)
+func parseMemoryString(memStr string) float64 {
+	if memStr == "" || memStr == "0" {
+		return 0.0
+	}
+
+	// Parse using Kubernetes resource.Quantity
+	quantity, err := resource.ParseQuantity(memStr)
+	if err != nil {
+		return 0.0
+	}
+
+	// Convert to GB
+	return quantity.AsApproximateFloat64() / (1024 * 1024 * 1024)
+}
+
+func processCpuModels(nodes []corev1.Node) bool {
+	cpuModelCounts := make(map[string]int)
+	cpuModelCores := make(map[string]*resource.Quantity)
+	cpuModelMemory := make(map[string]*resource.Quantity)
+
+	// Calculate cluster totals for normalization
+	totalNodes := len(nodes)
+	var totalCpu float64
+	var totalMemory float64
+	for _, node := range nodes {
+		if cpu := node.Status.Capacity.Cpu(); cpu != nil {
+			totalCpu += cpu.AsApproximateFloat64()
+		}
+		if mem := node.Status.Capacity.Memory(); mem != nil {
+			totalMemory += mem.AsApproximateFloat64() / (1024 * 1024 * 1024) // Convert to GB
+		}
+	}
+
+	for _, node := range nodes {
+		cpuModels := extractCpuModels(node.Labels)
+		memoryCapacity := node.Status.Capacity.Memory()
+		cpuCapacity := node.Status.Capacity.Cpu()
+
+		for _, cpuModel := range cpuModels {
+			cpuModelCounts[cpuModel]++
+
+			if cpuCapacity != nil {
+				if cpuModelCores[cpuModel] == nil {
+					cpuModelCores[cpuModel] = resource.NewQuantity(0, resource.DecimalSI)
+				}
+				cpuModelCores[cpuModel].Add(*cpuCapacity)
+			}
+
+			if memoryCapacity != nil {
+				if cpuModelMemory[cpuModel] == nil {
+					cpuModelMemory[cpuModel] = resource.NewQuantity(0, resource.BinarySI)
+				}
+				cpuModelMemory[cpuModel].Add(*memoryCapacity)
+			}
+		}
+	}
+
+	newHash := generateHash(cpuModelCounts, cpuModelCores, cpuModelMemory)
+
+	cpuModelInfo.lock.Lock()
+	defer cpuModelInfo.lock.Unlock()
+
+	if cpuModelInfo.lastUpdateHash == newHash {
+		return false
+	}
+
+	cpuModelInfo.totalNodes = totalNodes
+	cpuModelInfo.totalCpu = totalCpu
+	cpuModelInfo.totalMemory = totalMemory
+	cpuModelInfo.models = make(map[string]*v1beta1.CpuModelInfo)
+	for cpuModel, count := range cpuModelCounts {
+		cpuStr := "0"
+		if cpuModelCores[cpuModel] != nil {
+			cpuStr = cpuModelCores[cpuModel].String()
+		}
+
+		memStr := "0"
+		if cpuModelMemory[cpuModel] != nil {
+			memStr = cpuModelMemory[cpuModel].String()
+		}
+
+		cpuModelInfo.models[cpuModel] = &v1beta1.CpuModelInfo{
+			Name:      cpuModel,
+			Benchmark: getCpuModelPassMarkScore(cpuModel),
+			Nodes:     count,
+			Cpu:       cpuStr,
+			Memory:    memStr,
+		}
+	}
+
+	cpuModelInfo.lastUpdateHash = newHash
+	return true
+}
+
+func extractCpuModels(labels map[string]string) []string {
+	cpuModels := sets.New[string]()
```

Comment:
Not sure we need a set here, since `labels` is a map, we can have more than one `CpuModelLabelPrefix + aSpecificModelName` key. Consider use a simple slice here.